### ISSUE TITLE
WooCommerce Shipping Bug

### ISSUE THREAD

**Issue by jonathanhauge** [RainyDayMedia/carbide-probes#43]
"We are running into a glitch with orders through WooCommrce.
 
We have had two orders now that the customer has selected UPS Ground, but it the system is not charging them for the shipping cost.
 
If you go into WooCommerce, look at orders 12735 (from this morning) and 12649 from yesterday."

**Comment by Toddses** [comment]
Order 12649 the issue here is that for some reason it include the UPS group shipping even though the customer used their shipping account. So the 0 charge on shipping is correct here, but the shipping method somehow got flummoxed.

This is not the case for 12735. It looks like the customer did indeed select ground shipping, but for some reason it was not included in the subtotal breakdown.

I am investigating both of these issues now.

**Comment by jonathanhauge** [comment]
Thanks for getting going on this Todd.
ᐧ

-- 
Jonathan Hauge


marketing - branding - design - communications - digital

*A 2012 BBB Eclipse Integrity Award Finalist*

937 / 490-4005
1360 Technology Court  |  Suite 300  |  Beavercreek, OH 45430
www.baldwincreative.com

Confidentiality Notice: This email message, including any attachments, is
for the sole use of the intended recipient(s) and may contain confidential
and privileged information. Any unauthorized review, use, disclosure or
distribution is prohibited. If you are not the intended recipient, please
contact the sender by reply e-mail and destroy all copies of the original
message.

**Comment by Toddses** [comment]
A third order with the issue from Oct 20. Order 12512

**Comment by Toddses** [comment]
So far it seems pretty random. Can't replicate the issue after a couple hours of attempts.

**Comment by jonathanhauge** [comment]
@Toddses...

Here are some more.

 
"I just want to let you know that we received two more orders where the customer selected UPS Ground shipping, but the shipping charges were not charged.
 
The orders are 13078 and 12995."

"
I just received another online order without the shipping charges.  Order #13124."

**Comment by Toddses** [comment]
@jonathanhauge 

Okay I have a guess here. There is some issues with the Multiple Shipping Addresses and the UPS shipping plugins, and selecting ship to a different address on the checkout page. I have been able to sort of replicate it by playing with the shipping address fields. Essentially what I see is that even though Ground is selected, the shipping cost is not looped into the total cost. However, when I actually submit the order, the shipping cost is added in at that point.

So having said that, my guess is there's some magic combination that causes that cost to not get looped in on order submission.

Also that said, I've seen some other issues with the Multiple Shipping Address plugin.

**TLDR;** I'm going to disable the Multiple Shipping Addresses plugin. Please communicate that with Carbide and ask that if they see any other orders with this issue over the next week, to let us know ASAP. If we can confirm the issue lies here, we are in a much better position to come up with an alternate solution to saving multiple shipping addresses.

@ginfuru 

I'm going to open a support ticket with WooThemes and see if they've seen the issue. Maybe I can work with them to come up with a fix.

**Comment by ginfuru** [comment]
@Toddses 
I got the email for the support request. Thanks for the heads up... Though I think this is probably related to the customer's accounts and or Carbide not managing the shipping accounts correctly... remember all those special use cases for their operation?? 

@jonathanhauge Can you find out *what kind of customers* are having the shipping charges not come through on the orders... are these returning customers that have special pricing or what not.

**Comment by jonathanhauge** [comment]
@Toddses @ginfuru 

Here's some information from Jessica regarding this.

**Order# 13124** had a different Billing and Shipping address, no discount or special pricing, checked out as guest
 
**Order# 13078** had a different billing and shipping address, no discount or special pricing, created account
 
**Order#12995** same billing and shipping address, no discount or special pricing, created account
 
**Order# 12735** same billing and shipping address, no discount or special pricing, checked out as guest
 
**Order# 12649** same billing and shipping address, no discount or special pricing, created an account

**Comment by Toddses** [comment]
@jonathanhauge 

Thanks. Just let us know if any more orders come in with the issue.

**Comment by Toddses** [comment]
Thanks to HotJar we did actually get a recording of order 12649.

Basically what happened was after the user (who was a guest) entered their shipping information, **no options for UPS shipping showed up**. When there is enough billing information to get shipping rates, they are supposed to be returned and listed. This did not happen here. The only option was the custom shipping account (which the user eventually entered a number - their purchase order number, **not a valid shipping account**).

At this point the user enters their CC info, checks out, and the order appears to be a success. However, UPS Ground is listed as the shipping method, even though **this was never given as an option to the user**. This is why the user is not charged, because they never made an actual selection.

In the back end, there is no shipping added into the final cost, even though UPS Ground is listed as the shipping method, with the correct cost associated. It is just not added in.

So the issue appears to boil down to rates from UPS **not being listed in the shipping options** and yet UPS Ground is still somehow selected.

This at least gives us a jumping off point.

**Comment by Toddses** [comment]
@jonathanhauge @ginfuru 

Okay I have been able to replicate the issue! And I believe I was right previously, but for the wrong reasons.

You can as a guest, cause the issue with the billing address alone. You can, as a logged in user cause the issue with the separate shipping address form (since your billing info is usually already saved).

You must have multiple parts in your cart. It can be multiple different parts, or just more than 1 on the quantity of a part. You just need to be able to trigger the "You can ship to multiple addresses" message on the checkout page.

From there, you can futz around with the relevant form fields (state and zip), causing UPS Ground to be selected, and no charge being added to the order (even though it has the correct rate listed). You can then complete the order and you get no charge for shipping even though you may have had UPS Ground selected. I still need to do some work to come up with the **exact** steps for reproduction, which I will update with when I can lay those down.

That said, after I deactivate the Ship to Multiple Addresses plugin, I can no longer reproduce the issue. So this may indeed be an issue with that plugin, but just for different reasons than I thought.

@jonathanhauge That said, the action here is to keep the plugin deactivated until we can confirm that the issue is no longer occurring. Have carbide keep us posted if they notice an order with the issue. I'll keep an eye out myself, but since they get email notifications, they'll probably see it before me.

**Comment by Toddses** [comment]
OK I can reproduce this now easily. I think the problem lies in how WC selects a shipping method when you trigger the shipping method updates. For some reason it always wants to have UPS Ground selected initially, but it doesn't update the total cost correctly when the User Shipping Account plugin is activated (that's our custom plugin).

Currently looking into the cart-shipping template to see if we can resolve the issue there. There may also be a conflict with Multiple Shipping Address plugin not allowing WooCommerce to trigger an update to the total cost when the shipping information is fully entered.

**Comment by Toddses** [comment]
@ginfuru @jonathanhauge Okay I have nailed down the issue and can reproduce it 100% of the time with the method I'll outline below. It is indeed a bug in the Ship to Multiple Addresses plugin. We've got that open ticket with WooThemes so I've given them the steps as well, hopefully they can fix it.

That said, I don't think we should use the plugin. There's several other issues I am seeing with it. Even if this one is fixed, we're still going to have issues.

Here are the steps, just for future reference:

Enable WooCommerce (latest version), Enable Ship To Multiple Addresses (latest as well). WooCommerce Shipping Methods "Free Shipping" and "Local Delivery" should both be enabled. Give local delivery a fixed rate so you can see that it is not adding in the shipping to the total, and give it a specific zip code so you can control when its available. For the Selection Priorities, you want to set Local Delivery to 1 and Free Shipping to 2.

1. You can be logged in or acting as a guest.
2. Add products to your cart. You want to trigger the "You may use multiple shipping addresses on this cart" message during checkout, so you should add more than one product.
3. Proceed to checkout, and fill out your billing information so you trigger the local delivery method to be available.
4. Delete your zip code.
5. Select the free shipping option, so that the methods are updated, and free shipping is the only option available.
6. Reenter the local delivery zip code, triggering the available shipping methods to be updated again.
7. Local Delivery will be selected by default. The shipping rate will be correct, but the shipping cost will not be added in to your total cost.
8. You can now check out without being charged for shipping.

**Comment by Toddses** [comment]
okay this is pretty much resolved. @jonathanhauge this does mean users can no longer save multiple addresses. I will look for a free plugin for this for an easy and free solution.

**Comment by ginfuru** [comment]
:+1:
